compiler/hsSyn/Convert.hs

```diff
@@ -1512,6 +1513,12 @@ cvtTypeKind ty_str ty
            _ -> failWith (ptext (sLit ("Malformed " ++ ty_str)) <+> text (show ty))
     }
 
+hsTypeToArrow :: LHsType GhcPs -> HsArrow GhcPs
+hsTypeToArrow w = case unLoc w of
+                     HsTyVar _ _ (L _ n) | n == nameRdrName oneDataConName -> HsLinearArrow
+                     HsTyVar _ _ (L _ n) | n == nameRdrName omegaDataConName -> HsUnrestrictedArrow
+                     _ -> HsExplicitMult w
```

Comment:
@goldfirere says: use `isExact_maybe` and equality on `Name`s to avoid being stringly typed.